scripts/src/public/services/offers.ts

```diff
@@ -47,21 +48,30 @@ function offerDbToApi(offer: db.Offer, content: db.OfferContent) {
 	};
 }
 
+/**
+ * return the sublist of offers from this app that the user can complete
+ */
 async function filterOffers(userId: string, app: Application | undefined, logger: LoggerInstance): Promise<Offer[]> {
 	// TODO: this should be a temp fix!
 	// the app should not be undefined as we used left join, figure it out
 	if (!app) {
 		return [];
 	}
 
+	if (app.offers.length === 0) {
+		return [];
+	}
+	const offerCounts = await Order.countAllByOffer(userId);
+	const contents = await offerContents.getAllContents();
+
 	return (await Promise.all(
 		app.offers
 			.map(async offer => {
-				if (await offer.didExceedCap(userId)) {
+				if ((offerCounts.get(offer.id) || 0) >= await offer.cap.per_user) {
```

Comment:
`await offer.cap.per_user`?
it's not async

Reply:
you're right. Weird that it didn't complain